src/Container/CommandMiddlewareFactory.php

```diff
@@ -33,7 +34,8 @@ public function __invoke(ContainerInterface $container)
 
         return new CommandMiddleware(
             $container->get($options['command_bus']),
-            $container->get($options['message_factory'])
+            $container->get($options['message_factory']),
+            $container->get($options['metadata_gatherer'])
```

Comment:
To avoid the ZF\SM config dependency we could also make the "metadata_gatherer" option optional and use this line instead: `isset($options['metadata_gatherer'])? $container->get($options['metadata_gatherer']) : new NoopMetadataGatherer()`